### ISSUE TITLE
Underscore prefix not allowed for field name

### ISSUE THREAD

**Issue by rexxars** [gridsome/gridsome#351]
## Description

When adding a node with fields that start with an underscore (eg; `_key`, `_type`), these are stripped. This can cause conflicts and confusion when the node also has a non-prefixed attribute of the same name (`key` and `type` in this case).

### Steps to reproduce

```js
addNode({id: 'foo', fields: {_type: 'bar'}})
```

### Expected result

I expect my field names to be preserved. In general, I think camelcasing the input fieldnames is an odd behavior - totally valid GraphQL field names are given new names, which might break any consumer which relies on the data shape.

### Actual result

The keys are camelcased, prefix is removed.
I suppose this is the offending code: https://github.com/gridsome/gridsome/blob/3d7c180be776a582504d6c0099945d810748df06/gridsome/lib/app/ContentTypeCollection.js#L252-L258

I know this might be considered a breaking change, but perhaps we could at least opt-in to a non-field-normalization behavior when calling `addNode` or similar?

### Environment

<pre><code>
Libs:
- gridsome version: 0.5.6
</code></pre>

**Comment by hjvedvik** [comment]
I agree with you. If you create a field named `my_field` you would expect that to be `my_field` in the schema too. We can probably add an option for content types to enable camel casing of field names. Then each plugin can enable it if it wants to.

I'm going to update all the core source plugins to reflect the changes in #346 before we publish v0.6. So we could try to add an option for camel casing in the same update.

**Comment by rexxars** [comment]
That sounds great!

**Comment by vorti2** [comment]
Good afternoon @hjvedvik, how is your timetable to implement the option  `camelCasedFieldNames` in the source plugin?

**Comment by hjvedvik** [comment]
@vorti2 The `camelCasedFieldNames` option was meant as a temporary workaround for people that wanted to update to 0.6 but keep their old GraphQL queries. It will probably be removed. We're working on APIs for letting plugins define their own schemas with custom resolvers etc and I think that will be a better way to let source plugins camel case their field names.

**Comment by vorti2** [comment]
@hjvedvik thanks for clarification.
